client/lib/upgrades/actions/domain-management.js

```diff
@@ -126,17 +125,17 @@ function fetchDomains( siteId ) {
 		} );
 	}
 
+	const domains = DomainsStore.getBySite( siteId );
+	if ( domains.isFetching || domains.hasLoadedFromServer ) {
```

Comment:
Hmm, I was copying a pattern from fetchDns/fetchNameservers. Maybe we should remove those too, as those won't be able to refresh too.